src/compiler/checker.ts

```diff
@@ -17627,6 +17641,10 @@ namespace ts {
                 if (!isExternalOrCommonJsModule(file)) {
                     mergeSymbolTable(globals, file.locals);
                 }
+                if (file.patternAmbientModules && file.patternAmbientModules.length) {
+                    (patternAmbientModules || (patternAmbientModules = [])).push(...file.patternAmbientModules);

```

Comment:
i remember we had issues with this pattern for diagnostics, and concat fared better there. consider using `patternAmbientModules  = concatenate(patternAmbientModules , file.patternAmbientModules)` instead.